agent/main.go

```diff
@@ -258,3 +245,36 @@ func main() {
 		return
 	}
 }
+
+// generateKubeConfig will create a CertificateSigningRequest for the host
+// its running on and once the CSR is approved it will fetch the Certificate
+// and create a kubeconfig which will be used then by the host reconciler
+func generateKubeConfig(logger logr.Logger, hostName, boostrapKubeConfigPath string) error {
+	logger.Info("creating host csr")
+	bootstrapClientConfig, err := registration.LoadRESTClientConfig(bootstrapKubeConfig)
+	if err != nil {
+		return err
+	}
+	bootstrapClient, err := clientset.NewForConfig(bootstrapClientConfig)
+	if err != nil {
+		return err
+	}
+	byohCSR := registration.ByohCSR{BootstrapClient: bootstrapClient}
```

Comment:
Does this do the job of what we had originally intended for the bootstrap authenticator? As in wait / reconcile until the CSR is approved?

What happens in case of host agent restart?

Reply:
> As in wait / reconcile until the CSR is approved?

yes, `csr.WaitForCertificate` will wait for 60 mins before timing out.

> What happens in case of host agent restart?

 `csr.RequestCertificate` takes care of the creation of the CSR and it skips creation if the CSR already exists. Currently, this check will fail on https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/util/certificate/csr/csr.go#L82 due the key mismatch. This will get handled with the private key being persisted across restarts.

Reply:
Thanks.